### PR TITLE
Show additional stats on canvas

### PR DESCRIPTION
**Pull request by AyaPK**
- Now shows a Pokémon's atk/def/speed/special on the canvas
- Now shows a Pokémon's PP for each move in the canvas
  - Adds base PP values for every move to Byte mappings to facilitate this
- Updates tests accordingly

image

### CODE REVIEW

File: src/integration/gen_1_pokemon.py [hunk 1]
```diff
@@ -19,12 +20,24 @@ class Pokemon:
...
     level: int
+    attack: int
+    defense: int
+    speed: int
+    special: int
     status: str
     type: str
     move1: str
+    move1_pp: int
+    move1_max_pp: int
```
Comment (josephcwalker) [inline src/integration/gen_1_pokemon.py:31]: dunno how much you care about clean code but you _could_ put this in a struct
then have an array of moves

don't particularly care for it though

File: src/integration/gen_1_pokemon.py [hunk 2]
```diff
@@ -37,14 +50,30 @@ def from_memory_buffer(cls, nickname: str, caught_by: str, buffer) -> Self:
...
+            attack=int.from_bytes(buffer[0x24 : 0x24 + 2]),
+            defense=int.from_bytes(buffer[0x26 : 0x26 + 2]),
+            speed=int.from_bytes(buffer[0x28 : 0x28 + 2]),
+            special=int.from_bytes(buffer[0x2A : 0x2A + 2]),
             status=cls._status_from_bit_field(buffer[0x4]),
             type=cls._type_from_bytes(buffer[0x5], buffer[0x6]),
             move1=GEN_1_MOVES[buffer[0x8]],
+            move1_pp=(int(buffer[0x1D]) & 0x3F),
+            move1_max_pp=MOVE_BASE_PP[GEN_1_MOVES[buffer[0x8]]]
+            + ((int(buffer[0x1D]) >> 6) & 0x03),
```
Comment (josephcwalker) [inline src/integration/gen_1_pokemon.py:62]: what's the adding?

**Review by josephcwalker — APPROVED** [review]

File: src/integration/gen_1_pokemon.py (hunk 2 above)
Comment (AyaPK) [reply src/integration/gen_1_pokemon.py:62]: It takes the top two bits from the PP value for that move, which will be 00, 01, 10, or 11, depending on how many PP ups have been used. This is added to the base max PP. (The bottom 6 bits are the current PP value for that move).

File: src/integration/gen_1_pokemon.py (hunk 1 above)
Comment (AyaPK) [reply src/integration/gen_1_pokemon.py:31]: Yeah I honestly cba with that ahahaha

**Pull request merged**